orttraining/orttraining/core/graph/gradient_builder_base.cc

```diff
@@ -59,27 +59,33 @@ void ComputeBroadcastBackwardAxes(
       auto A_dim = A_dims[i].dim_param(),
            B_dim = B_dims[j].dim_param();
       if (A_dim != B_dim) {
-        ORT_THROW("Gradient building error for node ", node_name, ": symbolic dimension doesn't match. ",
-                  "A_dims:", ToString(A_dims), ", B_dims:", ToString(B_dims));
+        LOGS_DEFAULT(WARNING) << "Gradient building error for node " << node_name << ": symbolic dimension doesn't match. " <<
+                  "A_dims:" << ToString(A_dims) << ", B_dims:" << ToString(B_dims); 
       }
     } else if (A_dims[i].has_dim_param() && B_dims[j].has_dim_value()) {
       auto A_dim = A_dims[i].dim_param();
       auto B_dim = B_dims[j].dim_value();
 
       if (B_dim != 1) {
-        ORT_THROW("Gradient building error for node ", node_name, ": symbolic broadcasting requires the B_dimension to be 1. ",
-                  "A_dims:", ToString(A_dims), ", B_dims:", ToString(B_dims));
+        LOGS_DEFAULT(WARNING) << "Gradient building error for node " << node_name << ": symbolic broadcasting requires the B_dimension to be 1. " <<
+                  "A_dims:" << ToString(A_dims) << ", B_dims:" << ToString(B_dims);  
+        --i;
+        --j;            
+        continue;
       }
       if (B_axes) {
         B_axes->push_back(gsl::narrow_cast<int64_t>(k));
       }
     } else if (A_dims[i].has_dim_value() && B_dims[j].has_dim_param()) {
-      auto A_dim = A_dims[j].dim_value();
-      auto B_dim = B_dims[i].dim_param();
+      auto A_dim = A_dims[i].dim_value();
+      auto B_dim = B_dims[j].dim_param();
 
       if (A_dim != 1) {
-        ORT_THROW("Gradient building error for node ", node_name, ": symbolic broadcasting requires the A_dimension to be 1. ",
-                  "A_dims:", ToString(A_dims), ", B_dims:", ToString(B_dims));
+        LOGS_DEFAULT(WARNING) << "Gradient building error for node " << node_name << ": symbolic broadcasting requires the A_dimension to be 1. " <<
```

Comment:
please update the warning message to reflect the reality: this is a relaxing case, and the kernel might run into problem latter if there a_dims and b_dims turns out not boradcastable.